llvm/lib/Target/AMDGPU/AMDGPULowerModuleLDSPass.cpp

```diff
@@ -920,6 +920,124 @@ class AMDGPULowerModuleLDS {
     return KernelToCreatedDynamicLDS;
   }
 
+  static GlobalVariable *uniquifyGVPerKernel(Module &M, GlobalVariable *GV,
```

Comment:
This looks like an unrelated, separate change

Reply:
I will add you to the internal review link. So you can see the review history.

Reply:
This still should be a separate PR

Reply:
This is where the local pointer address is assigned for a named barrier before going into AMDGPUbackend

Reply:
This goes together with the change in AMDGPUMachineFunction.cpp

Reply:
The backend and frontend parts should still be split. Description also does not say how or what the modification is